stock_picking_auto_create_lot_qty/models/stock_picking.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (C) 2023 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _set_auto_lot(self):
+        """
+        Allows to be called either by button or through code
+        """
+        pickings = self.filtered(lambda p: p.picking_type_id.auto_create_lot)
+        lines = pickings.mapped("move_line_ids").filtered(
```

Comment:
I think better use one cycle for preparing value for `lines` and `new_lines` variables.
```py
lines = self.env[...].browse()
new_lines = self.env[...].browse()
for line in pickings.mapped("move_line_ids"):
    if not line.lot_id and not line.lot_name and line.product_id.tracking != 'none' and line.product.auto_create_lot:
        lines |= line
    if line.qty_done:
        new_lines |= line
````
This solution use only one cycle and your solution use two cycle.

Reply:
Thanks for hints) I'll fix it